sourcefiles/treasures/treasurewriter.py

```diff
@@ -128,6 +133,48 @@ def get_treasure_tier_dict(settings: rset.Settings):
     #       Let's just handle difficulty by moving boxes up/down in tier
     return treasure_tier_dict
 
+# if rset.GameFlags.TREASURE_TIER_MARKERS in settings.gameflags:
+def write_treasure_tier_markers(ct_rom: ctrom.CTRom,
+                                settings: rset.Settings):
+                
+    treasure_tier_dict = get_treasure_tier_dict(settings)
+    treasure_data = tt.get_base_treasure_dict()
+    treasure_marker_dict = td.get_treasures_tier_marker_dict()
+
+    for tier,marker in treasure_marker_dict.items():
+        treasures = treasure_tier_dict[tier]
+
+        for tid in treasures:
+            chest = treasure_data[tid]
+            chest_data = chest.get_chest_data(ct_rom)
+            location_id = tt.get_chest_loc_id(chest.chest_index)
+
+            # get script
+            script = ct_rom.script_manager.get_script(location_id)
+            obj_id = script.append_empty_object()
+            x_coord = (chest_data.x_coord << 4) + 10
+            y_coord = (chest_data.y_coord << 4) + 10
+
+            script.set_function(
+            obj_id, 0,
+            EF()
+            .add(EC.load_npc(marker)) # Use NPC as Marker
```

Comment:
At minimum, the load should be conditional on the treasure flag being unset.  Ideally, there's an active loop that will despawn the object after the treasure flag is hit.  You can mostly mimic this: https://github.com/Pseudoarc/jetsoftime/blob/b0f952754ca4fa941cf6e48e6f9ec510edf7ecbd/sourcefiles/objectivetypes.py#L780

As in the comments on that file, it would be useful to test rooms with many boxes to make sure that the running loops aren't slowing the scene down.  If there's significant performance degradation, there are some things we can try.